examples/custom_file_format/main.rs

```diff
@@ -0,0 +1,69 @@
+use config::{Config, Format, Map, Value, ValueKind, FileStoredFormat};
+use std::io::{Error, ErrorKind};
+
+/// The private and public key sources will be read into their associated variable:
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct Settings {
+    pub private_key: Option<String>,
+    pub public_key: Option<String>,
+}
+
+fn main() {
+    // Sourcing from two separate files for the `Settings` struct,
+    // these are both marked as optional to avoid failure if the file doesn't exist.
+    let file_public_key = config::File::new("examples/custom_file_format/files/public.pem", PemFile).required(false);
+    let file_private_key = config::File::new("examples/custom_file_format/files/private.pem", PemFile).required(false);
+
+    // Provide the sources and build the config object:
```

Comment:
```suggestion
    // Provide the sources and build the config object:
    // Both are marked as optional to avoid failure if the file doesn't exist.
```